finagle/h2/src/test/scala/com/twitter/finagle/buoyant/h2/netty4/Netty4StreamTransportTest.scala

```diff
@@ -351,31 +349,24 @@ class Netty4StreamTransportTest extends FunSuite {
   test("client: response cancelation resets stream: Failure(Reset)") {
     val ctx = new ClientCtx {}
     assert(!ctx.rspF.isDefined)
-    ctx.rspF.raise(Failure(Reset.EnhanceYourCalm))
+    ctx.rspF.raise(Reset.EnhanceYourCalm)
     assert(await(ctx.rspF.liftToTry) == Throw(Reset.EnhanceYourCalm))
   }
 
-  test("client: response cancelation resets stream: Failure(Interrupted)") {
+  test("client: response cancelation resets stream: Exception") {
     val ctx = new ClientCtx {}
     assert(!ctx.rspF.isDefined)
-    ctx.rspF.raise(Failure("foo").flagged(Failure.Interrupted))
+    ctx.rspF.raise(Failure("foo"))
     assert(await(ctx.rspF.liftToTry) == Throw(Reset.Cancel))
   }
 
   test("client: response cancelation resets stream: Failure(Rejected)") {
     val ctx = new ClientCtx {}
     assert(!ctx.rspF.isDefined)
-    ctx.rspF.raise(Failure("foo").flagged(Failure.Rejected))
+    ctx.rspF.raise(Reset.Refused)
     assert(await(ctx.rspF.liftToTry) == Throw(Reset.Refused))
   }
 
-  test("client: response cancelation resets stream: Exception") {
-    val ctx = new ClientCtx {}
-    assert(!ctx.rspF.isDefined)
-    ctx.rspF.raise(new Exception("ugh"))
-    assert(await(ctx.rspF.liftToTry) == Throw(Reset.InternalError))
```

Comment:
Reset.Cancel seems like a better default when a Future is interrupted than Reset.InternalError to me.